### PR TITLE
GPIO updates

### PR DESCRIPTION
**Pull request by amiclaus**
- add `gpio_get_optional` function
- ad9528: update gpio reset handling

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

### CODE REVIEW

File: drivers/frequency/ad9528/ad9528.c [hunk 1]
```diff
@@ -380,12 +380,13 @@ int32_t ad9528_setup(struct ad9528_dev **device,
...
-	if(init_param.hw_reset_en) {
-		ret = gpio_get(&dev->gpio_resetb, &init_param.gpio_resetb);
-		if (ret < 0)
-			return ret;
+	ret = gpio_get_optional(&dev->gpio_resetb, init_param.gpio_resetb);
+	if (ret < 0)
+		return ret;
+
+	if (dev->gpio_resetb)
 		ad9528_reset(dev);
```
[the remark this answers is not in the document]
Comment (amiclaus) [reply drivers/frequency/ad9528/ad9528.c:388]: I think it is done in the next line:
https://github.com/analogdevicesinc/no-OS/blob/37df58822266ea2aea17e7477ab1efb9bb984a69/drivers/frequency/ad9528/ad9528.c#L390-L394

[the remark this answers is not in the document]
Comment (amiclaus) [reply drivers/frequency/ad9528/ad9528.c:388]: But indeed, it is done also in the `ad9528_reset` function.

File: drivers/frequency/ad9528/ad9528.c [hunk 2]
```diff
@@ -380,12 +380,13 @@ int32_t ad9528_setup(struct ad9528_dev **device,
...
-	}
+
 	ret = ad9528_spi_write_n(dev,
```
Comment (buha) [inline drivers/frequency/ad9528/ad9528.c:390]: I like the new gpio_get_optional !
However, this soft reset (command over SPI) should be done only if no HW reset was done. It doesn't harm to also do it when a hard reset was done, but it doesn't make sense. So I simply suggest to move it in an else branch of the if (dev->gpio_resetb) condition. ;)

File: drivers/frequency/ad9528/ad9528.c [hunk 3]
```diff
@@ -380,12 +380,13 @@ int32_t ad9528_setup(struct ad9528_dev **device,
 		return ret;
 
 	/* GPIO */
-	if(init_param.hw_reset_en) {
-		ret = gpio_get(&dev->gpio_resetb, &init_param.gpio_resetb);
-		if (ret < 0)
-			return ret;
+	ret = gpio_get_optional(&dev->gpio_resetb, init_param.gpio_resetb);
+	if (ret < 0)
```
Comment (adrimbarean) [inline drivers/frequency/ad9528/ad9528.c:384]: Should the driver use the error.h definitions?

**Comment by amiclaus** [comment]
V2:
- ad9528: move soft reset from `ad9528_setup` to `ad9528_reset`

File: drivers/frequency/ad9528/ad9528.c (hunk 3 above)
Comment (dbogdan) [reply drivers/frequency/ad9528/ad9528.c:384]: Anyhow, I guess this is the subject of a different commit/pull-request.

File: drivers/frequency/ad9528/ad9528.c (hunk 2 above)
Comment (dbogdan) [reply drivers/frequency/ad9528/ad9528.c]: Without some additional testing, I would still keep the software reset too.

**Pull request merged**